src/updaters/changelog-json.ts

```diff
@@ -0,0 +1,104 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {ConventionalCommit} from '../commit';
+import {logger as defaultLogger, Logger} from '../util/logger';
+import {DefaultUpdater, UpdateOptions} from './default';
+import {randomUUID} from 'crypto';
+
+const BREAKING_CHANGE_TITLE = 'BREAKING CHANGE';
+const COMMIT_PREFIX = /^[^:]+: ?/;
+
+interface ChangelogJsonOptions extends UpdateOptions {
+  artifactName: string;
+  language: string;
+  commits: ConventionalCommit[];
+}
+
+interface Change {
+  type: string;
+  scope?: string;
+  sha: string;
+  message: string;
+  breakingChangeNote?: string;
+}
+
+/**
+ * Maintians a machine readable CHANGELOG in chnagelog.json.
+ * See: https://gist.github.com/bcoe/50ef0a0024bbf107cd5bc0adbdc04758
+ */
+export class ChangelogJson extends DefaultUpdater {
+  artifactName: string;
+  language: string;
+  commits: ConventionalCommit[];
+
+  /**
+   * Instantiate a new SamplesPackageJson updater
+   * @param options
+   */
+  constructor(options: ChangelogJsonOptions) {
+    super(options);
+    this.language = options.language;
+    this.artifactName = options.artifactName;
+    this.commits = options.commits;
+  }
+
+  /**
+   * Given initial file contents, return updated contents.
+   * @param {string} content The initial content
+   * @returns {string} The updated content
+   */
+  updateContent(content: string, logger: Logger = defaultLogger): string {
+    const parsed = JSON.parse(content);
+    logger.info(`adding release ${this.version} for ${this.artifactName}`);
+    const changes = [];
+    for (const commit of this.commits) {
+      // The commit.message field contains the type/scope prefix.
+      const message = commit.message.replace(COMMIT_PREFIX, '');
+      const change: Change = {
+        type: commit.type,
+        sha: commit.sha,
+        message: message,
+      };
+      if (commit.scope) change.scope = commit.scope;
+      for (const note of commit.notes) {
+        if (note.title === BREAKING_CHANGE_TITLE) {
+          change.breakingChangeNote = note.text;
+        }
+      }
+
+      // Including chores in changelog.json would cause the file to grow
+      // too quickly, as it would include the chore of creating the
+      // previous release itself (so all paths would always be updated).
+      if (commit.type === 'chore' && !change.breakingChangeNote) continue;
```

Comment:
I don't think this is the updater's job to determine whether to include these or not -- this feels like something the strategy should be deciding/customizing.

Reply:
It feels like significantly less toil to do it here, since we'd instead have to do it in every single strategy that has changelog.json support.

Reply:
I could see having a shared function that could filter commits by type - something like:

```ts
function filterCommits(commits: ConventionalCommit[], allowedTypes: string[], includeAnyBreakingChange: boolean = true): ConventionalCommit[] {}
```

Another option could be to include the filter options in the constructor for this updater. But having a hard-coded business decision in this class doesn't feel like the right place for that configuration.

Reply:
sounds reasonable, I'll just start with the logic in the Java Mono Repo strategy for now.